target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch

```diff
@@ -0,0 +1,154 @@
+ensures a higher possibility that the switch IC will have these pins back
+to the bootstrapping configuration before reset deassertion.
+
+With a cable from an active peer connected to port 3 before reset, the
+correct crystal frequency (0b10 = 40MHz) is selected:
+
+                      [1]        [2-1]     [3]     [5]
+                      :          :         :       :
+              _____________________________         __________________
+ESW_P4_LED_0                               |_______|
+              ___________________           _______
+ESW_P3_LED_0                     |_________|       |__________________
+
+                       :          :       : :     :
+                       :          [2-2]...: [4]...:
+                       [2]................:
+
+[1] Reset is asserted.
+[2] Period of 5000 - 5100 usec.
+[2-1] ESW_P3_LED_0 goes low.
+[2-2] Remaining period of 5000 - 5100 usec.
+[3] Reset is deasserted.
+[4] Period of 310 usec. HWTRAP register is populated with bootstrapped
+    XTAL frequency.
+[5] Signals reflect the bootstrapped configuration.
+
+ESW_P3_LED_0 low period before reset deassertion:
+
+              5000 usec
+            - 5100 usec
+    TEST     RESET HOLD
+       #         (usec)
+  ---------------------
+       1           5410
+       2           5440
+       3           4375
+       4           5490
+       5           5475
+       6           4335
+       7           4370
+       8           5435
+       9           4205
+      10           4335
+      11           3750
+      12           3170
+      13           4395
+      14           4375
+      15           3515
+      16           4335
+      17           4220
+      18           4175
+      19           4175
+      20           4350
+
+     Min           3170
+     Max           5490
+
+  Median       4342.500
+     Avg       4466.500
+
+Revert commit 2920dd92b980 ("net: dsa: mt7530: disable LEDs before reset").
+Changing the state of pins via reset assertion is simpler and more
+efficient than doing so by setting the LED controller off.
+
+Fixes: b8f126a8d543 ("net-next: dsa: add dsa support for Mediatek MT7530 switch")
+Fixes: c288575f7810 ("net: dsa: mt7530: Add the support of MT7531 switch")
+Co-developed-by: Justin Swartz <justin.swartz@risingedge.co.za>
+Signed-off-by: Justin Swartz <justin.swartz@risingedge.co.za>
+Signed-off-by: Arınç ÜNAL <arinc.unal@arinc9.com>
+Signed-off-by: David S. Miller <davem@davemloft.net>
+---
+ drivers/net/dsa/mt7530.c | 6 ------
+ 1 file changed, 6 deletions(-)
+
+--- a/drivers/net/dsa/mt7530.c
++++ b/drivers/net/dsa/mt7530.c
+@@ -2219,12 +2219,6 @@ mt7530_setup(struct dsa_switch *ds)
+ 		}
+ 	}
+ 
+-	/* Disable LEDs before reset to prevent the MT7530 sampling a
+-	 * potentially incorrect HT_XTAL_FSEL value.
+-	 */
+-	mt7530_write(priv, MT7530_LED_EN, 0);
+-	usleep_range(1000, 1100);
+-
+ 	/* Reset whole chip through gpio pin or memory-mapped registers for
+ 	 * different type of hardware
+ 	 */
```

Comment:
I would just remove this patch and 790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch.

Reply:
I like that all patches are now directly generated via `git format-patch` and are literally almost unchanged upstream commits. Removing that patch means also having to modify the follow-up patch which combines the revert with applying the alternative approach you were in favor of.

Reply:
The follow-up patch being "790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch"? You wouldn't have to modify that but just get rid of it. Currently, I see two patch files for a single patch. 789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch and 790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch. Another option should be to remove 789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch.

Reply:
Thanks for pointing that out.

I'll kept only the `789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch` and removed the redundant `790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch` and `790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch`